terraform-plans/modules/GitHub/settings/main.tf

```diff
@@ -0,0 +1,87 @@
+terraform {
+  required_providers {
+    github = {
+      source  = "integrations/github"
+      version = "~> 6.0"
+    }
+  }
+}
+
+provider "github" {
+  owner = var.owner
+  app_auth {} # using environmet variables for auth
+}
+
+resource "github_repository" "repo" {
+  name = var.repository
+
+  has_issues      = true
+  has_projects    = false
+  has_wiki        = false
+  has_discussions = false
+
+  allow_merge_commit = false
+  allow_squash_merge = true
+  allow_rebase_merge = false
+  allow_auto_merge   = false
+
+  allow_update_branch    = true
+  delete_branch_on_merge = true
```

Comment:
Did we discuss this one during Madrid engineering sprint? If not, please set it as false (default value in github).

Reply:
I am fine with this one. No reason to keep the merged branch.

Reply:
I'm ok with both, people often forgot to delete a branch, and then we have the lot of not relevant branches.

Reply:
Maybe leave it as an option (i.e. `false`); enabling it will block people from trying different development pattern (e.g. working on a dev branch for several small features, etc. No that we will do very often though)

Reply:
Please set it as "false" in this PR. We can change it after the discussion in the team.

Reply:
Done.